src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationExtensions.cs

```diff
@@ -19,10 +19,10 @@ public static class OAuthAuthenticationExtensions
         /// <param name="app">The <see cref="IApplicationBuilder"/> passed to the configure method.</param>
         /// <param name="options">The middleware configuration options.</param>
         /// <returns>The updated <see cref="IApplicationBuilder"/>.</returns>
-        public static IApplicationBuilder UseOAuthAuthentication([NotNull] this IApplicationBuilder app, [NotNull] string authenticationScheme, Action<OAuthAuthenticationOptions<IOAuthAuthenticationNotifications>> configureOptions = null)
+        public static IApplicationBuilder UseOAuthAuthentication([NotNull] this IApplicationBuilder app, [NotNull] string authenticationScheme, Action<OAuthAuthenticationOptions<IOAuthAuthenticationNotifications<OAuthAuthenticatedContext>, OAuthAuthenticatedContext>> configureOptions = null)
         {
-            return app.UseMiddleware<OAuthAuthenticationMiddleware<OAuthAuthenticationOptions<IOAuthAuthenticationNotifications>, IOAuthAuthenticationNotifications>>(
-                new ConfigureOptions<OAuthAuthenticationOptions<IOAuthAuthenticationNotifications>>(options =>
+            return app.UseMiddleware<OAuthAuthenticationMiddleware<OAuthAuthenticationOptions<IOAuthAuthenticationNotifications<OAuthAuthenticatedContext>, OAuthAuthenticatedContext>, IOAuthAuthenticationNotifications<OAuthAuthenticatedContext>, OAuthAuthenticatedContext>>(
```

Comment:
It looks far better when completely removing the generic `IOAuthAuthenticationNotifications` parameters: https://github.com/PinpointTownes/Security/commit/511334d4726e1aedd6deb0ed4240d1c85fe586eb